torch/csrc/jit/python/init.cpp

```diff
@@ -122,6 +124,23 @@ using ::c10::FunctionSchema;
 using caffe2::serialize::PyTorchStreamReader;
 using caffe2::serialize::PyTorchStreamWriter;
 
+class PythonSymbolicIntNode: public c10::SymbolicIntNode {
+public:
+  PythonSymbolicIntNode(py::object pyobj): 
+    c10::SymbolicIntNode(), 
+    pyobj_(std::move(pyobj)) {};
+
+  virtual SymbolicIntNode* add(SymbolicIntNode* other) override {
+    auto pother = dynamic_cast<PythonSymbolicIntNode*>(other);
+    py::gil_scoped_acquire acquire;
+    auto r = pyobj_.attr("__add__")(pother->pyobj_);
```

Comment:
@ezyang this works for now, but maybe there's a smarter way to do redispatch.

Reply:
this probably will get overhauled when we get more data from @Chillee about what kind of API. Calling magic methods on the ugly inner object is weird, probably just have them be normal methods.

Reply:
Yeah, it does look weird.